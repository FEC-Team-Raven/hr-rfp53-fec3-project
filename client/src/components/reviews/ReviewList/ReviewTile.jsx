import React, {useState, useEffect} from 'react';
import Stars from '../../Stars.jsx';
import axios from 'axios';


const ReviewTile = (props) => {
  const [helpfulness, setHelpfullness] = useState(props.review.helpfulness);
  const [voted, setVoted] = useState(false);
  const [expandBody, setExpandBody] = useState(false);
  const [displayImgModal, setDisplayImgModal] = useState(false);
  const [modalImage, setModalImage] = useState('');

  useEffect(() => {
    setHelpfullness(props.review.helpfulness);
  }, [props.review]);

  var renderDate = (date) => {
    var date = new Date(date);
    var options = {year: 'numeric', month: 'long', day: 'numeric' };
    return date.toLocaleDateString('en-US', options);
  };

  var renderRecommend = recommend => {
    if (recommend) {
      return <div>
        <i className="fas fa-check" style={{'margin-right': '5px'}}></i>
        I recommend this product
      </div>;
    }
  };

  var renderResponse = response => {
    if (response) {
      return (
        <div className="response">
          Response: <br/>
          {response}
        </div>
      );
    }
  };

  var renderBody = body => {
    if (body.length > 250 && !expandBody) {
      var display = body.slice(0, 250) + '...';
      return (
        <div>
          {display} <br />
          <a className="showMore" onClick={() => setExpandBody(true)}>Show More</a>
        </div>
      );
    }
    return <div>{body}</div>;
  };

  var renderImageModal = () => {
    if (displayImgModal) {
      return (
        <div className="imageModal">
          <span className="close" onClick={() => setDisplayImgModal(false)}>X</span>
          <img className="modal-content" src={modalImage}></img>
        </div>
      );
    }
  };

  var updateModal = url => {
    setModalImage(url);
    setDisplayImgModal(true);
  };

  var postHelpfulness = () => {
    if (!voted) {
      setVoted(true);
      axios({
        url: 'http://localhost:3000/reviews/helpful',
        method: 'PUT',
        params: {
          reviewId: props.review.review_id,
        }
      })
        .then(() => {
          setHelpfullness(helpfulness + 1);
        })
        .catch(err => {
          console.log(err);
        });
    }
  };

  var reportReview = () => {
    axios({
      url: 'http://localhost:3000/reviews/report',
      method: 'PUT',
      params: {
        reviewId: props.review.review_id
      }
    });
  };

  return (
    <div className="reviewTile">
      <div className="reviewRow between">
        <Stars rating={props.review.rating} />
        <div className="rating">{props.review.reviewer_name}, {renderDate(props.review.date)}</div>
      </div>
      <div className="summary">{props.review.summary}</div>
      {renderBody(props.review.body)}
      {renderRecommend(props.review.recommend)}
      {renderResponse(props.review.response)}
      {props.review.photos.map(photo => {
        return <img className="reviewImage" src={photo.url} onClick={() => updateModal(photo.url)}></img>;
      })}
      {renderImageModal()}
      <div className="reviewRow">
        Helpful?
        <button onClick={() => postHelpfulness()} style={{'margin-left': '5px'}}>Yes ({helpfulness})</button>
        |
        <button onClick={reportReview}>Report</button>
      </div>
    </div>
  );
};

export default ReviewTile;
