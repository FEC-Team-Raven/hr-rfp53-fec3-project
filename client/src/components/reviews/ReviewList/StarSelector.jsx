import React, {useState} from 'react';

const StarSelector = (props) => {
  const [rating, setRating] = useState(1);
  const [clicked, setClicked] = useState(false);

  var ratingToWidth = rating => {
    rating = Math.floor(rating / 0.25) * 0.25;
    var decimal = rating - Math.floor(rating);
    rating *= 20;
    if (decimal === 0.25) {
      rating += 3;
    } else if (decimal === 0.75) {
      rating -= 3;
    }
    return rating;
  };

  var handleHover = event => {
    console.log(event.target.getAttribute('data-star'));
    if (!clicked) {
      setRating(event.target.getAttribute('data-star'));
    }
  };

  return (
    <div className="star-selector-container">
      <div className="star-selector-rating reviewRow" style={{'width': `${ratingToWidth(rating)}%`}}>
        <div data-star="1" onMouseOver={handleHover}>&#x2605;</div>
        <div data-star="2" onMouseOver={handleHover}>&#x2605;</div>
        <div data-star="3" onMouseOver={handleHover}>&#x2605;</div>
        <div data-star="4" onMouseOver={handleHover}>&#x2605;</div>
        <div data-star="5" onMouseOver={handleHover}>&#x2605;</div>
      </div>
      <div className="starOverlay reviewRow">
        <div data-star="1" className="emptyStar" onMouseOver={handleHover}></div>
        <div data-star="2" className="emptyStar" onMouseOver={handleHover}></div>
        <div data-star="3" className="emptyStar" onMouseOver={handleHover}></div>
        <div data-star="4" className="emptyStar" onMouseOver={handleHover}></div>
        <div data-star="5" className="emptyStar" onMouseOver={handleHover}></div>
      </div>
      <div className="star-selector-holo reviewRow">
        <div>&#x2606;</div>
        <div>&#x2606;</div>
        <div>&#x2606;</div>
        <div>&#x2606;</div>
        <div>&#x2606;</div>
      </div>
    </div>
  );
};

export default StarSelector;