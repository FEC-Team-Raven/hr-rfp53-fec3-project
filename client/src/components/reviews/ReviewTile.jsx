import React, {useState, useEffect} from 'react';
import Stars from '../Stars.jsx';
import axios from 'axios';


const ReviewTile = (props) => {
  const [helpfulness, setHelpfullness] = useState(props.review.helpfulness);
  const [voted, setVoted] = useState(false);
  const [expandBody, setExpandBody] = useState(false);
  const [displayModal, setDisplayModal] = useState(false);
  const [modalImage, setModalImage] = useState('');

  useEffect(() => {
    setHelpfullness(props.review.helpfulness);
  }, [props]);

  var renderDate = (date) => {
    var date = new Date(date);
    var options = {year: 'numeric', month: 'long', day: 'numeric' };
    return date.toLocaleDateString('en-US', options);
  };

  var renderRecommend = recommend => {
    if (recommend) {
      return <div>
        <i className="fas fa-check" style={{'margin-right': '5px'}}></i>
        I recommend this product
      </div>;
    }
  };

  var renderResponse = response => {
    if (response) {
      return (
        <div className="response">
          Response: <br/>
          {response}
        </div>
      );
    }
  };

  var renderBody = body => {
    if (body.length > 250 && !expandBody) {
      var display = body.slice(0, 250) + '...';
      return (
        <div>
          {display} <br />
          <a className="showMore" onClick={() => setExpandBody(true)}>Show More</a>
        </div>
      );
    }
    return <div>{body}</div>;
  };

  var renderImageModal = () => {
    if (displayModal) {
      return (
        <div className="imageModal">
          <span className="close" onClick={() => setDisplayModal(false)}>X</span>
          <img className="modal-content" src={modalImage}></img>
        </div>
      );
    }
  };

  var updateModal = url => {
    setModalImage(url);
    setDisplayModal(true);
  };

  var postHelpfulness = () => {
    console.log('HELLO');
    if (!voted) {
      console.log('HELLO');
      setVoted(true);
      axios({
        url: 'http://localhost:3000/reviews/helpful',
        method: 'PUT',
        params: {
          reviewId: props.review.review_id,
        }
      })
        .then(() => {
          setHelpfullness(helpfulness + 1);
        })
        .catch(err => {
          console.log(err);
        });
    }
  };

  return (
    <div className="reviewTile">
      <div className="reviewRow between">
        <Stars rating={props.review.rating} />
        <div>{props.review.reviewer_name}, {renderDate(props.review.date)}</div>
      </div>
      {renderBody(props.review.body)}
      {renderRecommend(props.review.recommend)}
      {renderResponse('TEST RESPONSE')}
      {props.review.photos.map(photo => {
        return <img className="reviewImage" src={photo.url} onClick={() => updateModal(photo.url)}></img>;
      })}
      {renderImageModal()}
      <div className="reviewRow">
        Helpful?
        <button onClick={() => postHelpfulness()} style={{'margin-left': '5px'}}>Yes ({helpfulness})</button>
      </div>
    </div>
  );
};

export default ReviewTile;
