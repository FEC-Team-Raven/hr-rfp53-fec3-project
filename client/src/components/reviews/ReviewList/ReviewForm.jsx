import React from 'react';
import StarSelector from './StarSelector.jsx';

const ReviewForm = props => {

  var ratingToWidth = rating => {
    rating = Math.floor(rating / 0.25) * 0.25;
    var decimal = rating - Math.floor(rating);
    rating *= 20;
    if (decimal === 0.25) {
      rating += 3;
    } else if (decimal === 0.75) {
      rating -= 3;
    }
    return rating;
  };


  //<span>&#x2605;&#x2605;&#x2605;&#x2605;&#x2605;</span>
  return (
    <div className="reviewFormModal">
      <span className="close" onClick={() => props.setDisplayFormModal(false)}>X</span>
      <div className="reviewFormContainer">
        <div className="starContainer">
          <StarSelector />
        </div>
      </div>
    </div>
  );
};

export default ReviewForm;
