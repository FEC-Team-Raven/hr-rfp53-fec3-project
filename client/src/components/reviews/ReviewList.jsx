import React, {useState, useContext} from 'react';

import {ReviewContext} from './Reviews.jsx';
import ReviewTile from './ReviewTile.jsx';

const ReviewList = () => {
  const reviews = useContext(ReviewContext);

  return (
    <div>
      {reviews.length} reviews, sorted by
      <select>
        <option value="relevance">relevance</option>
        <option value="helpful">helpful</option>
        <option value="newest">newest</option>
      </select> <br />
      {reviews.map((review) => <ReviewTile review={review} />)}
      <button>More Reviews</button>
      <button>Add a Review +</button>
    </div>
  );
};

export default ReviewList;
