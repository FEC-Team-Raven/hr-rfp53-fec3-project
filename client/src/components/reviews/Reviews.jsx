import React, {useState, useEffect} from 'react';
import ReviewList from './ReviewList/ReviewList.jsx';
import ProductBreakdown from './ProductBreakdown/ProductBreakdown.jsx';
import axios from 'axios';

export const ReviewContext = React.createContext({});

const Reviews = (props) => {
  const [loading, setLoading] = useState(true);

  var context = {
    starFilter: useState([]),
    metaData: useState({}),
  };

  useEffect(() => {
    if (props.productId) {
      getMetaData();
    }
  }, [props.productId]);

  var getMetaData = () => {
    axios('http://localhost:3000/reviews/meta', {
      params: {
        productId: props.productId
      }
    })
      .then(meta => {
        context.metaData[1](meta.data);
        setLoading(false);
      });
  };

  context.getMetaData = getMetaData;

  if (!loading) {
    return (
      <div className={`module ${props.theme}-3 ${props.theme}-text`} id="reviews">
        RATINGS & REVIEWS
        <div className="ratingsReviewsContainer">
          <ReviewContext.Provider value={context}>
            <ProductBreakdown productId={props.productId} theme={props.theme}/>
            <ReviewList productId={props.productId} theme={props.theme}/>
          </ReviewContext.Provider>
        </div>
      </div>
    );
  }
  return null;
};

export default Reviews;
