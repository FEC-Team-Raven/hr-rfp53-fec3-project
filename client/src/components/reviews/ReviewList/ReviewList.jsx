import React, {useState, useEffect, useContext} from 'react';
import axios from 'axios';
import ReviewTile from './ReviewTile.jsx';
import ReviewForm from './ReviewForm/ReviewForm.jsx';
import {ReviewContext} from '../Reviews.jsx';

const ReviewList = (props) => {
  const [reviews, setReviews] = useState([]);
  const [renderedReviews, setRenderedReviews] = useState(reviews);
  const [sort, setSort] = useState('relevant');
  const [displayCount, setDisplayCount] = useState(2);
  const [displayFormModal, setDisplayFormModal] = useState(false);
  const [searchFilter, setSearchFilter] = useState('');
  var starFilter = useContext(ReviewContext).starFilter[0];

  useEffect(() => {
    if (props.productId !== 0) {
      getReviews();
    }
  }, [props.productId, sort]);


  useEffect(() => {
    filter(reviews);
  }, [starFilter, searchFilter]);

  var filter = reviews => {
    if (starFilter.length === 0 && searchFilter.length < 3) {
      setRenderedReviews(reviews);
    } else if (searchFilter.length < 3) {
      setRenderedReviews(reviews.filter(review => {
        return starFilter.includes(review.rating);
      }));
    } else if (starFilter.length === 0) {
      setRenderedReviews(reviews.filter(review => {
        return review.body.toLowerCase().includes(searchFilter.toLowerCase()) || review.summary.toLowerCase().includes(searchFilter.toLowerCase());
      }));
    } else {
      setRenderedReviews(reviews.filter(review => {
        return starFilter.includes(review.rating);
      })
        .filter(review => {
          return review.body.toLowerCase().includes(searchFilter.toLowerCase()) || review.summary.toLowerCase().includes(searchFilter.toLowerCase());
        }));
    }
  };

  var getReviews = () => {
    return axios('http://localhost:3000/reviews', {
      params: {
        productId: props.productId,
        sort: sort
      }
    })
      .then(reviews => {
        setReviews(reviews.data.results);
        filter(reviews.data.results);
      })
      .catch(err => {
        console.log(err);
      });
  };

  var handleClick = () => {
    setDisplayCount(displayCount + 2);
  };

  var renderMoreReviews = () => {
    if (displayCount < reviews.length) {
      return <button id="moreReviews" className={`reviewButton ${props.theme}-2 ${props.theme}-text`} onClick={handleClick}>MORE REVIEWS</button>;
    }
  };

  var renderList = () => {
    if (reviews.length > 0) {
      return (
        <div className="scrollable">
          {renderedReviews.slice(0, displayCount).map((review, index) => <ReviewTile key={index} review={review} theme={props.theme}/>)}
        </div>
      );
    }
  };

  var renderReviewFormModal = () => {
    if (displayFormModal) {
      return <ReviewForm productId={props.productId} setDisplayFormModal={setDisplayFormModal} getReviews={getReviews} theme={props.theme}/>;
    }
  };

  return (
    <div className="reviewList">
      {reviews.length} reviews, sorted by
      <select className={`sortReviews ${props.theme}-3 ${props.theme}-text`} onChange={e => setSort(e.target.value)}>
        <option value="relevance">relevance</option>
        <option value="helpful">helpful</option>
        <option value="newest">newest</option>
      </select>
      <input
        className={`${props.theme}-1 ${props.theme}-text`}
        type="text"
        onChange={e => setSearchFilter(e.target.value)}
      /> <br />

      {renderList()}
      {renderMoreReviews()}
      <button className={`reviewButton ${props.theme}-2 ${props.theme}-text`} onClick={() => setDisplayFormModal(true)}>ADD A REVIEW   +</button>
      {renderReviewFormModal()}
    </div>
  );
};

export default ReviewList;
