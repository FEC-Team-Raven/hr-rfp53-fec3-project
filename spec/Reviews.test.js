/* eslint-disable camelcase */
import React from 'react';
import renderer from 'react-test-renderer';
import Reviews from '../client/src/components/reviews/Reviews.jsx';
import ReviewList from '../client/src/components/reviews/ReviewList/ReviewList.jsx';
import ReviewTile from '../client/src/components/reviews/ReviewList/ReviewTile.jsx';

it('renders Reviews correctly', () => {
  const tree = renderer
    .create(<Reviews />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders ReviewTile correctly', () => {
  var testReview = {
    body: 'I hate cucumber pls dont throw it at me i see a bird i stare at it i meow at it i do a wiggle come here birdy look at dog hiiiiiisssss. Pet my belly, you know you want to; seize the hand and shred it! spend all night ensuring people don\'t sleep sleep all day, claws in the eye of the beholder so attack feet, ignore the human until she needs to get up, ',
    date: '2021-04-20T00:00:00.000Z',
    helpfulness: 8,
    photos: [],
    rating: 3,
    recommend: true,
    response: null,
    review_id: 347986,
    reviewer_name: 'JoeCool',
    summary: 'Gate keepers of hell.'
  };

  const tree = renderer
    .create(<ReviewTile review={testReview} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders ReviewTile response if there is a response correctly', () => {
  var testReview = {
    body: 'I hate cucumber pls dont throw it at me i see a bird i stare at it i meow at it i do a wiggle come here birdy look at dog hiiiiiisssss. Pet my belly, you know you want to; seize the hand and shred it! spend all night ensuring people don\'t sleep sleep all day, claws in the eye of the beholder so attack feet, ignore the human until she needs to get up, ',
    date: '2021-04-20T00:00:00.000Z',
    helpfulness: 8,
    photos: [],
    rating: 3,
    recommend: true,
    response: 'TEST RESPONSE',
    review_id: 347986,
    reviewer_name: 'JoeCool',
    summary: 'Gate keepers of hell.'
  };

  const tree = renderer
    .create(<ReviewTile review={testReview} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});