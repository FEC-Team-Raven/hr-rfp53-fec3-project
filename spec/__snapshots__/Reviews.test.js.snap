// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CharacteristicBreakdown correctly 1`] = `
<div
  className="characteristicContainer"
>
  <div>
    Fit
  </div>
  <div>
    <div
      className="characteristicBar"
    />
  </div>
  <div
    className="pointer"
    style={
      Object {
        "left": "242.99999999999997px",
      }
    }
  >
    <i
      className="fas fa-sort-up"
    />
  </div>
  <div
    className="reviewRow characteristic"
  >
    <div>
      Too tight
    </div>
    <div>
      Perfect
    </div>
    <div>
      Too loose
    </div>
  </div>
</div>
`;

exports[`renders CharacteristicInput correctly 1`] = `
<div
  className="characteristicInputContainer"
>
  <div
    style={
      Object {
        "text-decoration": "underline",
      }
    }
  >
    Comfort
  </div>
  <div
    className="characteristicInput"
  >
    <div
      className="grid"
    >
      <div>
        <input
          id="Comfort1"
          name="Comfort"
          onChange={[Function]}
          required={true}
          type="radio"
          value={1}
        />
        <label
          for="Comfort1"
        >
          Uncomfortable
        </label>
      </div>
    </div>
    <div
      className="grid"
    >
      <div>
        <input
          id="Comfort2"
          name="Comfort"
          onChange={[Function]}
          required={true}
          type="radio"
          value={2}
        />
        <label
          for="Comfort2"
        >
          Slightly uncomfortable
        </label>
      </div>
    </div>
    <div
      className="grid"
    >
      <div>
        <input
          id="Comfort3"
          name="Comfort"
          onChange={[Function]}
          required={true}
          type="radio"
          value={3}
        />
        <label
          for="Comfort3"
        >
          Ok
        </label>
      </div>
    </div>
    <div
      className="grid"
    >
      <div>
        <input
          id="Comfort4"
          name="Comfort"
          onChange={[Function]}
          required={true}
          type="radio"
          value={4}
        />
        <label
          for="Comfort4"
        >
          Comfortable
        </label>
      </div>
    </div>
    <div
      className="grid"
    >
      <div>
        <input
          id="Comfort5"
          name="Comfort"
          onChange={[Function]}
          required={true}
          type="radio"
          value={5}
        />
        <label
          for="Comfort5"
        >
          Perfect
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders ReviewTile correctly 1`] = `
<div
  className="reviewTile"
>
  <div
    className="reviewRow between"
  >
    <span>
      <div
        className="star-rating"
        style={
          Object {
            "width": "60%",
          }
        }
      >
        <span>
          ★★★★★
        </span>
      </div>
    </span>
    <div
      className="rating"
    >
      JoeCool
      , 
      April 19, 2021
    </div>
  </div>
  <div
    className="summary"
  >
    Gate keepers of hell.
  </div>
  <div>
    I hate cucumber pls dont throw it at me i see a bird i stare at it i meow at it i do a wiggle come here birdy look at dog hiiiiiisssss. Pet my belly, you know you want to; seize the hand and shred it! spend all night ensuring people don't sleep sleep...
     
    <br />
    <a
      className="showMore"
      onClick={[Function]}
    >
      Show More
    </a>
  </div>
  <div>
    <i
      className="fas fa-check"
      style={
        Object {
          "margin-right": "5px",
        }
      }
    />
    I recommend this product
  </div>
  <div
    className="reviewRow"
  >
    Helpful?
    <button
      onClick={[Function]}
      style={
        Object {
          "margin-left": "5px",
        }
      }
    >
      Yes (
      8
      )
    </button>
    |
    <button
      onClick={[Function]}
    >
      Report
    </button>
  </div>
</div>
`;

exports[`renders ReviewTile response if there is a response correctly 1`] = `
<div
  className="reviewTile"
>
  <div
    className="reviewRow between"
  >
    <span>
      <div
        className="star-rating"
        style={
          Object {
            "width": "60%",
          }
        }
      >
        <span>
          ★★★★★
        </span>
      </div>
    </span>
    <div
      className="rating"
    >
      JoeCool
      , 
      April 19, 2021
    </div>
  </div>
  <div
    className="summary"
  >
    Gate keepers of hell.
  </div>
  <div>
    I hate cucumber pls dont throw it at me i see a bird i stare at it i meow at it i do a wiggle come here birdy look at dog hiiiiiisssss. Pet my belly, you know you want to; seize the hand and shred it! spend all night ensuring people don't sleep sleep...
     
    <br />
    <a
      className="showMore"
      onClick={[Function]}
    >
      Show More
    </a>
  </div>
  <div>
    <i
      className="fas fa-check"
      style={
        Object {
          "margin-right": "5px",
        }
      }
    />
    I recommend this product
  </div>
  <div
    className="response"
  >
    Response: 
    <br />
    TEST RESPONSE
  </div>
  <div
    className="reviewRow"
  >
    Helpful?
    <button
      onClick={[Function]}
      style={
        Object {
          "margin-left": "5px",
        }
      }
    >
      Yes (
      8
      )
    </button>
    |
    <button
      onClick={[Function]}
    >
      Report
    </button>
  </div>
</div>
`;

exports[`renders StarSelector correctly 1`] = `
<div
  className="star-selector-container"
>
  <div
    className="star-selector-rating reviewRow"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    <div>
      ★★★★★
    </div>
  </div>
  <div
    className="starOverlay reviewRow"
  >
    <div
      className="emptyStar"
      data-star={1}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="emptyStar"
      data-star={2}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="emptyStar"
      data-star={3}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="emptyStar"
      data-star={4}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
    <div
      className="emptyStar"
      data-star={5}
      onClick={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    />
  </div>
  <div
    className="star-selector-holo reviewRow"
  >
    <div>
      ☆☆☆☆☆
    </div>
  </div>
</div>
`;
